sktime/forecasting/base/adapters/_neuralforecast.py

```diff
@@ -288,19 +319,18 @@ def _fit(
         # | Index                   | B2.2.1    |
         # | Index (Missing)         | B2.2.2    |
         # | Other                   | unreached |
-
+        y_time_index = y.index.get_level_values(-1)
         if self.freq != "auto":  # A: freq is given as non-auto
             self._freq = self.freq
         elif fh.freq:  # B1: freq is inferred from fh
             self._freq = fh.freq
-        elif isinstance(y.index, pandas.DatetimeIndex):  # B2.1: y is date-like
+        elif isinstance(y_time_index, pandas.DatetimeIndex):  # B2.1: y is date-like
             raise ValueError(
                 f"Error in {self.__class__.__name__}, could not interpret freq, try "
                 "passing freq in model initialization or use a valid offset in index"
             )
         else:  # B2.2: y is not date-like
-            diffs = np.unique(np.diff(y.index))
-
+            diffs = np.unique(np.diff(np.unique(y_time_index)))
```

Comment:
Question: Is the inner np.unique guaranteed to give sorted dates for multi-index? What happens if "first" series have missing rows and "second" does not?

Reply:
> Question: Is the inner `np.unique` guaranteed to give sorted dates, even for original multi-index?

Yes `np.unique` will sort the result: https://numpy.org/doc/stable/reference/generated/numpy.unique.html.



> What happens if "first" series have missing rows and "second" does not?

It's not considered as we only have one `fh` and one `cutoff`.  Maybe there are other method to fill the missing value to make all time series equal lenght?